polar-core/tests/integration_tests.rs

```diff
@@ -1133,12 +1169,16 @@ fn test_in_op() {
 #[test]
 fn test_matches() {
     let mut polar = Polar::new();
-    qnull(&mut polar, "x = 1 and y = 2 and x matches y");
-    qeval(&mut polar, "x = 1 and y = 1 and x matches y");
-
-    qeval(&mut polar, "x = {foo: 1} and x matches {foo: 1}");
-    qnull(&mut polar, "x = {foo: 1} and x matches {foo: 1, bar: 2}");
-    qnull(&mut polar, "x = {foo: 1} and x matches {foo: 2}");
+    assert!(qnull(&mut polar, "1 matches 2"));
+    assert!(qeval(&mut polar, "1 matches 1"));
+    // This doesn't fail because `y` is parsed as an unknown specializer
+    // assert!(qnull(&mut polar, "x = 1 and y = 2 and x matches y"));
```

Comment:
What's the goal with this comment? That the behavior will be fixed to allow this in the future?

Reply:
This was never functioning, but there was a non-checked test for it. I didn't want to outright remove it, because it does seem like an incorrect case.